### ISSUE TITLE
Use of hybrid-fuctional-based vbm/cbm in a unitcell

### ISSUE THREAD

**Issue by FPrith** [kumagai-group/pydefect#22]
My question is on the use of vbm and cbm values in a unitcell calculated using a hybrid fuctional.

Thanks to the Pydefect manual, PBEsol-based DFT calculations and defect analyses seemed to be successfully done in simple binary and ternary systems.
(Analyses described in the optional section, Section 12, were partially neglected)

To obtain correct vbm and cbm values in a unitcell, the band structure of the unitcell was evaluated using a hybrid functional.
This function was only applied to the unitcell; so, the total enegies of competing phases and perfect/defective supercells were evaluated using the PBEsol functional.

A pydefect des command and pydefect pe command generated a error as follows:
WARNING: Supercell CBM 11.5581 is lower in energy than the unitcell CBM 12.6293
In addtion, as described in an attached pdf file, this error was expressed as two dashed lines with "supercell VBM" and "supercell CBM" words. 
[energy_test_PBEsol_hybrid.pdf](https://github.com/user-attachments/files/18235577/energy_test_PBEsol_hybrid.pdf)

Were these situations problematic to obtain defect-formation energies?
In other words, could you tell me how to use hybrid-fuctional-based vbm/cbm values?

**Comment by yuuukuma** [comment]
There is a serious issue when you use the hybrid functional combining with PBEsol calcualtions, because the absolute VBM/CBM values cannot be directly compared.

**Comment by FPrith** [comment]
Thank you for your reply.

To my knowlegde, in your paper described below, structural relaxation and band-structure calculaions were done with the PBEsol functional and non-sc dd PBEsol0 hybrid functionals, respectively.
In addition, this paper described that defect-related processes were done with pydefect.
https://journals.aps.org/prapplied/abstract/10.1103/PhysRevApplied.19.034063

Is it impossible of doing those calculations with pydefect, in principle?

**Comment by yuuukuma** [comment]
Yes. If you use non-sc hybrid functional, you can use the VBM and CBM values because the wavefunctions are not updated.

**Comment by FPrith** [comment]
Thank you for your reply.

Fortunately, I have used non-sc dielectric-dependent hybrid functionals.
Your comment means that, in my cases, vbm/cbm information obtained in this type of functional can be used for point-defect analyses, doesn't it?

How should I use those vbm/cbm values in unitcells?
I guess one or more *.yaml files need to be modified by hand or in LINUX commands.

**Comment by yuuukuma** [comment]
You just change the unitcell.yaml.

**Comment by FPrith** [comment]
The pdf file "energy_test_PBEsol_hybrid.pdf", which was uploaded in my fist comment, was created according to your last comment; the corresponding unitcell.yaml file included nsc-dd-PBEsol0-based vbm and cbm values.
But, my analyses failed and/or were a little bit weird.
I think that something wrong happens in my yaml files, json.files, and DFT outputs.

I will do a defect analysis for one of the systems that are reported by your paper, and will judge whether my process is appropriate or not!
After this calculation, I will ask you again.
https://journals.aps.org/prapplied/abstract/10.1103/PhysRevApplied.19.034063

**Comment by FPrith** [comment]
I analyzed oxygen-vacancy-related properties in hexagonal ZnO, which was reported by this article, with an almost same DFT/pydefect inputs; as a result, oxygen-vacancy-formation energies were comparable to those in this article.
https://journals.aps.org/prmaterials/abstract/10.1103/PhysRevMaterials.5.123803

I believed that a non-sc dd hybrid functional was correctly used in my analyses.

Thank you very much for your kind indications!

**Comment by yuuukuma** [comment]
That's a great news! 
Thank you for sharing your calculation results.
